### PR TITLE
Get Tags Route

### PR DESCRIPTION
**Pull request by braydonwang**
## Notion task link

[Get Tags Route](https://www.notion.so/uwblueprintexecs/2792c3504d744d2b90ff967c32d00961?v=d33629815ecf477ab4e1ff4e33774c32&p=cddd8fc25bd441c598811798a7847550&pm=s)

## Implementation description
* created a new get tags route that returns a list of all tags from the db
* created a new get tags service and interface

## Steps to test
1. You can try running the backend locally and try calling the URL: http://localhost:8080/tags through postman (as an unauthed user, it should display: "error": "You are not authorized to make this request." and as an admin, it should display a list of tag objects)

## What should reviewers focus on?
* Focus on the architecture and the flow from controller to service layer

## Checklist
- [X] My PR name is descriptive and in imperative tense
- [X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [X] If I have made API changes, I have updated the [REST API Docs](https://www.notion.so/uwblueprintexecs/REST-Endpoints-05ce60312bb943439dfda42bb1318536)
- [] IF I have made changes to the db/models, I have updated the [Data Models Page](https://www.notion.so/uwblueprintexecs/Data-Models-760f8aa06b244eb0842c079ad77987b0)
- [ ] I have updated other Docs as needed

### CODE REVIEW

File: backend/app/rest/tag_routes.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from flask import Blueprint, current_app, jsonify
+from ..middlewares.auth import require_authorization_by_role
+from ..services.implementations.tags_service import TagsService
+
+tags_service = TagsService(current_app.logger)
```
Comment (owen-sellner) [inline backend/app/rest/tag_routes.py]: nit: For `tag_routes.py` and `tags_service.py` I think both names should have tags be plural

File: backend/app/services/implementations/tags_service.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        Create an instance of TagsService
+
+        :param logger: application's logger instance
+        :type logger: logger
+        """
+        self.logger = logger
+
+    def get_tags(self):
+        try:
+            tags_results = Tag.query.all()
```
Comment (owen-sellner) [inline backend/app/services/implementations/tags_service.py:21]: I do not know if it is an issue but I noticed that it is getting all tags including ones with the status "Deleted".

image

**Review by owen-sellner — CHANGES_REQUESTED** [review]
I commented about a small change but overall it was really well done!

File: backend/app/services/implementations/tags_service.py (hunk 2 above)
Comment (connor-bechthold) [reply backend/app/services/implementations/tags_service.py:21]: @owen-sellner this is fine for now, I've made a future ticket to remove this field
https://www.notion.so/uwblueprintexecs/2792c3504d744d2b90ff967c32d00961?v=d33629815ecf477ab4e1ff4e33774c32&p=58fb5fab828147748139026c4a4d81dc&pm=s

**Review by connor-bechthold — APPROVED** [review]
This works perfectly!!! Great work on getting this in quickly for your first PR 👨‍🍳 😎

**Review by owen-sellner — APPROVED** [review]
LGTM!!!!

**Review by phamkelly17 — APPROVED** [review]

**Pull request merged**